### ISSUE TITLE
Error: Cannot find module 'mocha/lib/cli/run-option-metadata'

### ISSUE THREAD

**Issue by sumukhah** [wix/Detox#2694]
Configuring Detox on android giving me this error on running `npx detox test --configuration android.emu.release`
* My .detoxrc is written similar to [this](https://github.com/wix/Detox/blob/master/docs/Introduction.Android.md)

Getting this errors 
<img width="907" alt="Screenshot 2021-03-19 at 8 56 11 PM" src="https://user-images.githubusercontent.com/23723464/111803925-8c786e80-88f5-11eb-8f54-8d0fd766870e.png">

**Comment by noomorph** [comment]
I'll check if Mocha has changed anything in their folder structure.

**Comment by noomorph** [comment]
What version of Mocha do you use? @sumukhah

**Comment by sumukhah** [comment]
I am not using Mocha at all.(In my project)

**Comment by noomorph** [comment]
@sumukhah , there's something wrong with your `testRunner` property in your Detox config then.

**Comment by sumukhah** [comment]
I have generated it through detox init -r jest. So I believe it's correct.

**Comment by sumukhah** [comment]
Yes. testRunner is set to : "jest-circus/runner" in e2e/config.json

**Comment by noomorph** [comment]
@sumukhah, actually I've been actually inquiring about your `detox.config.js` (or `.detoxrc.json`, ...), and not `e2e/config.json`.

**Comment by sumukhah** [comment]
Oh sorry...
My .detoxrc.json is 

```{
  "devices": {
    "emulator": {
      "type": "android.emulator",
      "device": {
        "avdName": "Pixel_API_29"
      }
    }
  },
  "apps": {
    "android.debug": {
      "type": "android.apk",
      "binaryPath": "android/app/build/outputs/apk/debug/app-debug.apk",
      "build": "cd android && ./gradlew assembleDebug assembleAndroidTest -DtestBuildType=debug && cd .."
    },
    "android.release": {
      "type": "android.apk",
      "binaryPath": "android/app/build/outputs/apk/release/app-release.apk",
      "build": "cd android && ./gradlew assembleRelease assembleAndroidTest -DtestBuildType=release && cd .."
    }
  },
  "configurations": {
    "android.emu.debug": {
      "device": "emulator",
      "app": "android.debug"
    },
    "android.emu.release": {
      "device": "emulator",
      "app": "android.release"
    }
  }
}

```

**Comment by noomorph** [comment]
@sumukhah , so this is exactly where I would expect to see:

```diff
 {
+  "testRunner": "jest",
+  "runnerConfig": "e2e/config.json",
   "devices": {
     "emulator": {
...
```

Due to compatibility reasons, default runner for Detox is `mocha`, although de facto we recommend `jest`.

**Comment by noomorph** [comment]
@sumukhah , could you run `which mocha` in your terminal, please?

**Comment by sumukhah** [comment]
No i haven't installed mocha globally
> mocha not found

**Comment by noomorph** [comment]
Thanks, now I get why I could not reproduce this issue. 👍 
I'll improve the error message soon.

**Comment by noomorph** [comment]
Action items for the pull request:

- [ ] print a warning about an undefined `testRunner` property
- [ ] throw an error if the test runner cannot be resolved (have you installed?...)

**Comment by shootermv** [comment]
i have same problem as @sumukhah 

after adding testRunner and runnerConfig rules to detoxrc.json
i started to get:
```
 Error [ERR_PACKAGE_PATH_NOT_EXPORTED]: Package subpath './build/types' is not defined by "exports" in C:\myProjects\localization-class\node_modules\jest-circus\package.json 
```

**Comment by noomorph** [comment]
@shootermv make sure you are using Detox 18.16.0 and above.

**Comment by d4vidi** [comment]
Closing (won't fix): We are working on removing mocha support, at least for the time being, in [Detox 20](https://github.com/wix/Detox/issues/3348).
